File: clients/csharp/src/MustelaLog.Client.Core/Configuration/ClientSettings.cs
namespace MustelaLog.Client.Core.Configuration;

/// <summary>Oberste Konfigurationsklasse für den Desktop-Client.</summary>
public sealed class ClientSettings
{
    public ApiSettings Api { get; set; } = new();
    public DiagnosticsSettings Diagnostics { get; set; } = new();
    public UiSettings Ui { get; set; } = new();
}

File: clients/java/src/main/java/de/sasd/mustelalog/client/config/ClientSettings.java
package de.sasd.mustelalog.client.config;

/**
 * Root configuration model for the desktop client.
 */
public final class ClientSettings
{
    private ApiSettings api = new ApiSettings();
    private UiSettings ui = new UiSettings();
    private DiagnosticsSettings diagnostics = new DiagnosticsSettings();

    public ApiSettings getApi() { return api; }
    public void setApi(ApiSettings api) { this.api = api; }
    public UiSettings getUi() { return ui; }
    public void setUi(UiSettings ui) { this.ui = ui; }
    public DiagnosticsSettings getDiagnostics() { return diagnostics; }
    public void setDiagnostics(DiagnosticsSettings diagnostics) { this.diagnostics = diagnostics; }
}
